### PR TITLE
Add CDT for libnl

### PR DESCRIPTION
**Pull request by jakirkham**
Adds a CDT package for `libnl`. Generated by running `conda skeleton rpm --distro centos6 libnl`.

Follows the discussion of packaging CDTs in conda-forge. ( https://github.com/conda-forge/conda-forge.github.io/issues/783 )

Checklist

- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vendor other packages
- [ ] Build number is 0
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by jakirkham** [comment]
Thought @conda-forge/core on next steps?

**Comment by jakirkham** [comment]
Also cc-ing @mingwandroid and @nehaljwani for awareness/comment.

**Comment by mingwandroid** [comment]
PRs welcome for the linter issues here. I'm a bit busy.

**Comment by scopatz** [comment]
Some of these are pretty interesting linter failures.

**Comment by scopatz** [comment]
This probably needs maintainers at the very least

**Comment by jakirkham** [comment]
Some of the errors seem to be the linter getting confused. Much of that metadata is specified for the `outputs`. That said, given there is only one output, maybe using `outputs` is unnecessary here. @mingwandroid, are there CDTs that do have multiple `outputs`? If not, how do you feel about streamlining this a bit by removing `outputs` from the template?

**Comment by nehaljwani** [comment]
Is this the first conda-forge-only-CDT-package PR?

**Comment by jakirkham** [comment]
AFAIAA yes.

**Comment by mingwandroid** [comment]
I cannot remember why I used outputs really! Don't think there's any need though.

**Comment by jakirkham** [comment]
Have added PR ( https://github.com/conda/conda-build/pull/3548 ) to rewrite the CDT template without `outputs`. This cuts out a fair amount of lint.

**Comment by jakirkham** [comment]
With the following PRs, I'm able to get rid of almost all lint. Everything else that is not addressed is discussed below.

* https://github.com/conda/conda-build/pull/3548 - fixes metadata lint
* https://github.com/conda/conda-build/pull/3549 - fixes build number lint
* https://github.com/conda/conda-build/pull/3551 - adds a basic test

Currently there doesn't appear to be a good way to checksum source RPMs as they lack validated checksums. Have raised issue ( https://github.com/conda/conda-build/issues/3552 ) to see if there are any alternatives.

Adding maintainers is generally on us to do. So I don't know if it is worth changing the template further to accommodate this.

Please let me know if I missed anything.

**Comment by jakirkham** [comment]
Hmm...seems Linux is running into a peculiar failure. Have raised as issue ( https://github.com/conda/conda-build/issues/3553 ).

**Comment by jakirkham** [comment]
@scopatz, could you give it another look please? I think this is about as close to making the linter happy as we can for now without some conda-build fixes.

**Comment by scopatz** [comment]
This looks good to me. Is it OK to merge?

File: recipes/libnl-cos6-x86_64/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package:
+  name: libnl-cos6-x86_64
+  version: 1.1.4
+
+source:
+  - url: http://mirror.centos.org/centos/6.10/os/x86_64/Packages/libnl-1.1.4-2.el6.x86_64.rpm
+    sha256: 03a7235e747695208a2633fd6515a39377a9b06b7aa866bda1a1e1eabf418869
+    folder: binary
+  - url: http://vault.centos.org/6.10/os/Source/SPackages/libnl-1.1.4-2.el6.src.rpm
```
Comment (isuruf) [inline recipes/libnl-cos6-x86_64/meta.yaml]: @jakirkham, can you add a sha256 for this?

Comment (jakirkham) [reply recipes/libnl-cos6-x86_64/meta.yaml]: Please see issue ( https://github.com/conda/conda-build/issues/3552 ).

**Comment by jakirkham** [comment]
> Is it OK to merge?

Yep, please feel free to merge.

File: recipes/libnl-cos6-x86_64/meta.yaml (hunk 1 above)
Comment (isuruf) [reply recipes/libnl-cos6-x86_64/meta.yaml]: That doesn't mean, it couldn't be added manually. See https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L19 for instructions

**Comment by jakirkham** [comment]
Adding the devel CDT associated with this in PR ( https://github.com/conda-forge/staged-recipes/pull/8633 ).

File: recipes/libnl-cos6-x86_64/meta.yaml (hunk 1 above)
Comment (jakirkham) [reply recipes/libnl-cos6-x86_64/meta.yaml]: I dropped this from the feedstock. It is unused AFAICT. ( https://github.com/conda-forge/libnl-cos6-x86_64-feedstock/pull/1 )

**Pull request merged**